### ISSUE TITLE
torch.nn.functional.nll_loss behaves differently in two cases of cpu and cuda

### ISSUE THREAD

**Issue by ilovepytorch** [pytorch/pytorch#66872]
## 🐛 Bug

`torch.nn.functional.nll_loss` works without any error when using cuda, but throws an error when using cpu.

## To Reproduce

```python
import torch
def cpu():
  arg_0 = torch.rand(torch.Size([500, 6]), dtype=torch.float32)
  arg_1 = torch.randint(-32768,32768,torch.Size([500]), dtype=torch.int64)
  res = torch.nn.functional.nll_loss(arg_0,arg_1,)
  
def cuda():
  arg_0 = torch.rand(torch.Size([500, 6]), dtype=torch.float32).cuda()
  arg_1 = torch.randint(-32768,32768,torch.Size([500]), dtype=torch.int64).cuda()
  res = torch.nn.functional.nll_loss(arg_0,arg_1,)
  
print(cuda())
print("CUDA PASS!")
print(cpu())
```

`IndexError: Target -1481 is out of bounds.` when using cpu

## Expected behavior

Then should have the same behavior.

## Environment

- PyTorch Version (e.g., 1.0): 1.9.1
- OS (e.g., Linux): Linux
- How you installed PyTorch (conda, pip, source): pip
- Python version: 3.8.10
- CUDA/cuDNN version: 11.3
- GPU models and configuration: 3080TI

cc @brianjo @mruberry @albanD @jbschlosser @walterddr

**Comment by jbschlosser** [comment]
Closing as this is fixed on master.

**Comment by malfet** [comment]
@ngimel do we need to cherry-pick the fix into 1.10.1?

**Comment by ngimel** [comment]
Yes, #67167 is milestoned, fix is #64954, and I've just milestoned it too.
